### PR TITLE
tap_manager: do not lock during tap creation

### PR DESCRIPTION
**Pull request by toanju**
during tap creation we still receive notifications from cnetlink. This
cannot be block since netlink messages might be dropped. fixes f1d021e

### CODE REVIEW

File: src/netlink/tap_manager.cpp [hunk 1]
```diff
@@ -153,18 +153,31 @@ tap_manager::~tap_manager() { destroy_tapdevs(); }
...
+      dev_name_exists = true;
+  }
+
+  if (!dev_exists && !dev_name_exists) {
     // create a new tap device
 
     ctapdev *dev;
     try {
       dev = new ctapdev(port_name);
       tap_devs.insert(std::make_pair(port_id, dev));
```
Comment (DFritzsche) [inline src/netlink/tap_manager.cpp]: operation on tap_devs is not locked, but above it is

Comment (toanju) [reply src/netlink/tap_manager.cpp]: good catch, so I will remove it above as well since this structure does not need to be locked.

Comment (DFritzsche) [reply src/netlink/tap_manager.cpp]: so l. 164/165 can be removed from lock as well, also this would improve code clarity

Comment (toanju) [reply src/netlink/tap_manager.cpp]: done

**Pull request merged**